### ISSUE TITLE
please add sparsemax function, tabnet need this, but hard to code

### ISSUE THREAD

**Issue by coco-boy** [tensorflow/tensorflow#71125]
### Issue type

Feature Request

### Have you reproduced the bug with TensorFlow Nightly?

Yes

### Source

binary

### TensorFlow version

tf 2.16

### Custom code

No

### Current behavior?

please add sparsemax function

### Standalone code to reproduce the issue

```shell
please add sparsemax function
```

### Relevant log output

```shell
please add sparsemax function
```

**Comment by tilakrayal** [comment]
@coco-boy,
Have you got the chance to have a look at tfa.layers.Sparsemax and tfa.activations.Sparsemax which will be used as layers and the activations. Kindly refer to the official document and let us know if you are looking for the similar feature.

https://www.tensorflow.org/addons/api_docs/python/tfa/layers/Sparsemax
https://www.tensorflow.org/addons/api_docs/python/tfa/activations/sparsemax

Thank you!

**Comment by coco-boy** [comment]
> @coco-boy, Have you got the chance to have a look at tfa.layers.Sparsemax and tfa.activations.Sparsemax which will be used as layers and the activations. Kindly refer to the official document and let us know if you are looking for the similar feature.
> 
> https://www.tensorflow.org/addons/api_docs/python/tfa/layers/Sparsemax https://www.tensorflow.org/addons/api_docs/python/tfa/activations/sparsemax
> 
> Thank you!

tfa project, This project is deprecated. TensorFlow Addons has stopped development, so hope to add the new feature

**Comment by tilakrayal** [comment]
@coco-boy,
 [Relu Activation Function](https://arxiv.org/pdf/1505.00853.pdf%C3%A3%E2%82%AC%E2%80%9AReLU) seems to be good for handling the problem of Vanishing and Exploding Gradients.

Can you please let us know how efficient are the activation functions that you proposed compared to the existing non-saturated activation functions like [Relu](https://www.tensorflow.org/api_docs/python/tf/keras/activations/relu), [Elu](https://www.tensorflow.org/api_docs/python/tf/keras/activations/elu), [Selu](https://www.tensorflow.org/api_docs/python/tf/keras/activations/selu)? Thank you!
